metric/bigtable/src/main/java/com/spotify/heroic/metric/bigtable/BigtableBackend.java

```diff
@@ -345,9 +368,12 @@ public Statistics getStatistics() {
 
         for (final Map.Entry<RowKey, Mutations.Builder> e : building.entrySet()) {
             final ByteString rowKeyBytes = rowKeySerializer.serializeFull(e.getKey());
-            writes.add(client
-                .mutateRow(table, rowKeyBytes, e.getValue().build())
-                .directTransform(result -> timer.end()));
+            final Mutations.Builder builder = e.getValue();
+            if (builder.size() > 0) {
```

Comment:
FWIW: This happens because line 354 creates an empty builder when it swaps out the batch. I think I'd prefer killing this section of code in favor of conditionally moving builder into the `saved` list to avoid this duplicate logic.

Does that make sense?